sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1926,7 +1926,7 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_COMMENT
-    : '/*' {!isHint()}? (BRACKETED_COMMENT|.)*? '*/' -> channel(HIDDEN)
+    : '/*' {!isHint()}? ( BRACKETED_COMMENT | . )*? ('*/' | EOF) -> channel(HIDDEN)
```

Comment:
I prefer to use a new channel, then we just need to check the last element in that channel, because we only want to error out for the EOF case.

This can avoid the expensive for loop to check all the elements in the hidden channel.